docs/rfcs/1020-pki-support.md

```diff
@@ -0,0 +1,291 @@
+- **Accepter**: A user/device admin of an organization in charge of accepting or not the join request.
+
+## Goals and Non-Goals
+
+Some prior work was already done to add some primitives (sign, verify, encrypt, decrypt) to work with the windows certificate store under the `libparsec_platform_pki`.
+
+The goals are to define the higher elements to make the PKI work with Parsec for the invitation process, those higher elements are:
+
+- [`pki_enrollment_select_certificate`]
+- [`pki_enrollment_sign_payload`]
+- [`pki_enrollment_create_local_pending`]
+- [`pki_enrollment_load_peer_certificate`]
+- [`pki_enrollment_load_submit_payload`]
+- [`pki_enrollment_load_accept_payload`]
+- [`pki_enrollment_load_local_pending_secret_part`]
+
+[`pki_enrollment_select_certificate`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L217-L251
+[`pki_enrollment_sign_payload`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L254-L276
+[`pki_enrollment_create_local_pending`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L279-L333
+[`pki_enrollment_load_peer_certificate`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L336-L355
+[`pki_enrollment_load_submit_payload`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L385-L406
+[`pki_enrollment_load_accept_payload`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L409-L430
+[`pki_enrollment_load_local_pending_secret_part`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L433-L473
+
+## Design
+
+Here is a possible user flow of Alice joining the organization CoolOrg using the PKI to enable the async enrollment feature:
+
+ 1. Alice wants to join CoolOrg organization.
+ 2. Alice already has a PKI configured on its system and the link to submit the join request.
+ 3. Alice select the certificate to use for the enrollment ([`pki_enrollment_select_certificate`])
+
+    > [!IMPORTANT]
+    > The certificate need to have a `CommonName` ([oid-2.5.4.3](https://oid-base.com/get/2.5.4.3)) and a `EmailAddress` ([oid-1.2.840.113549.1.9.1](https://oid-base.com/get/1.2.840.113549.1.9.1)) attributes in the certificate's subject.
+
+ 4. Alice creates an encryption (for the user certificate) and signing key pair (for the device certificate).
+ 4. Alice save the private parts of the key pairs on the filesystem ([`pki_enrollment_create_local_pending`]).
+ 5. Alice submits the request (authorized to re-submit later on with updated information) with: ([`pki_enrollment_submit`]):
+
+    - The PKI certificate.
```

Comment:
Should we add something like this?
```suggestion
    - The PKI certificate (from its smartcard or from the certificate store)
```

Reply:
It's a details not important to Parsec with the used API: https://github.com/Scille/parsec-cloud/pull/11231/files#r2394867382